poetry/core/masonry/builders/wheel.py

```diff
@@ -168,9 +168,20 @@ def _write_metadata(self, wheel):
             with self._write_to_zip(wheel, self.dist_info + "/entry_points.txt") as f:
                 self._write_entry_points(f)
 
+        license_files_to_add = []
         for base in ("COPYING", "LICENSE"):
-            for path in sorted(self._path.glob(base + "*")):
-                self._add_file(wheel, path, "%s/%s" % (self.dist_info, path.name))
+            license_files_to_add.append(self._path / base)
+            license_files_to_add.extend(self._path.glob(base + ".*"))
+
+        license_files_to_add.extend((self._path / "LICENSES").glob("**" + os.sep + "*"))
```

Comment:
```suggestion
        license_files_to_add.extend(self._path.glob("LICENSES/**/*"))
```
pathlib is smart enough to handle OS diferences.

Reply:
Useful to know, thanks!

Reply:
It turns out that this isn't the case when using the `.glob` method.  I'm not sure why it isn't normalised, but you can see in [this](https://github.com/python-poetry/poetry-core/runs/993745827) run when this change was added that the only failure was in the Windows builds.

I've reverted this change as a result.

Reply:
I'll check shortly but this might be related to https://bugs.python.org/issue31202

Reply:
Looks like it is. In this case; can you try this suggestion? This should also confirm that the issue is as identified above or if pathlib does not handle os diferences as advertised.

```suggestion
        license_files_to_add.extend(self._path.joinpath("LICENSES").glob("**/*"))
```

Reply:
It could be related - it's odd that it only affects things when the "incorrect" pathsep is used though!

Even if it were fixed, Poetry would still need to support the older versions where the fix hasn't been implemented yet, so I think that means we're stuck with `sep` for now.